cloud/scope/managed_machine_pool.go

```diff
@@ -343,6 +349,67 @@ func (m *ManagedMachinePoolScope) CreateNodePool(ctx context.Context) (*oke.Node
 	return m.getOKENodePoolFromOCID(ctx, nodePoolId)
 }
 
+func (m *ManagedMachinePoolScope) setNodepoolImageId(ctx context.Context) error {
+	imageId := m.OCIManagedMachinePool.Spec.NodeSourceViaImage.ImageId
+	if imageId != nil && *imageId != "" {
+		return nil
+	}
+	response, err := m.ContainerEngineClient.GetNodePoolOptions(ctx, oke.GetNodePoolOptionsRequest{
+		NodePoolOptionId: common.String("all"),
+		CompartmentId:    common.String(m.OCIManagedCluster.Spec.CompartmentId),
+	})
+	if err != nil {
+		m.Logger.Error(err, "Could not lookup node pool options")
+		return err
+	}
+	// version in the capoci spec starts with v, wheres as in the node pool options API in OKE
+	// the image name does not contain . For example
+	//       {
+	//        "image-id": "ocid1.image.oc1.iad.aaaaaaaafp7ysdbfl2bg67s4jzbpewxo3k772baixa3vwzecwogl474qecza",
+	//        "source-name": "Oracle-Linux-8.6-aarch64-2022.12.15-0-OKE-1.25.4-543",
+	//        "source-type": "IMAGE"
+	//      },
+	//      {
+	//        "image-id": "ocid1.image.oc1.iad.aaaaaaaauwwokidwf5nfi34ucbvgnjsni3klnfmu6pz73ctdohb2byiw6ztq",
+	//        "source-name": "Oracle-Linux-8.6-2022.12.15-0-OKE-1.23.4-543",
+	//        "source-type": "IMAGE"
+	//      },
+	// we will only default to non gpu image
+	k8sVersion := (*m.OCIManagedMachinePool.Spec.Version)[1:]
```

Comment:
Is it possible this gets an index out of bounds error?

Reply:
good point, but made me think that we dont have validation on this. Added validation webhook validation, and just to be sure, added vaidation here as well